File: app/src/main/java/fr/erictruong/material/training/ui/resources/ResourcesActivity.java
package fr.erictruong.material.training.ui.resources;

import android.support.v4.app.Fragment;
import android.content.Context;
import android.content.Intent;

import fr.erictruong.material.training.R;
import fr.erictruong.material.training.ui.core.BaseDrawerActivity;
import fr.erictruong.material.training.ui.core.DummyFragment;

public class ResourcesActivity extends BaseDrawerActivity {

    /**
     * Start an {@link ResourcesActivity} with extras.
     *
     * @param context The Android context.
     * @param id      (Extra) The navigation menu item ID of the desired content.
     */
    public static void start(Context context, int id) {
        Intent starter = new Intent(context, ResourcesActivity.class);
        starter.putExtra(EXTRA_NAV_ID, id);
        context.startActivity(starter);
    }

    @Override
    protected Fragment getSelectedFragment(int id) {
        switch (id) {
            case R.id.nav_color_palettes:
                return DummyFragment.newInstance();
            case R.id.nav_devices:
                return DummyFragment.newInstance();
            case R.id.nav_layout_templates:
                return DummyFragment.newInstance();
            case R.id.nav_roboto_and_noto_fonts:
                return DummyFragment.newInstance();
            case R.id.nav_sticker_sheets_and_icons:
                return DummyFragment.newInstance();
            default:
                return null;
        }
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/resources/ResourcesActivity.kt
package fr.erictruong.material.training.ui.resources

import android.support.v4.app.Fragment
import fr.erictruong.material.training.R
import fr.erictruong.material.training.ui.core.BaseDrawerActivity
import fr.erictruong.material.training.ui.resources.color_palettes.ColorPalettesFragment
import fr.erictruong.material.training.ui.resources.devices.DevicesFragment
import fr.erictruong.material.training.ui.resources.layout_templates.LayoutTemplatesFragment
import fr.erictruong.material.training.ui.resources.roboto_noto_fonts.RobotoNotoFontsFragment
import fr.erictruong.material.training.ui.resources.sticker_sheets_icons.StickerSheetsIconsFragment
import org.jetbrains.anko.AnkoLogger

class ResourcesActivity : BaseDrawerActivity() {

    override fun getSelectedFragment(id: Int): Fragment? {
        return when (id) {
            R.id.nav_resources_color_palettes -> ColorPalettesFragment()
            R.id.nav_resources_devices -> DevicesFragment()
            R.id.nav_resources_layout_templates -> LayoutTemplatesFragment()
            R.id.nav_resources_roboto_noto_fonts -> RobotoNotoFontsFragment()
            R.id.nav_resources_sticker_sheets_icons -> StickerSheetsIconsFragment()
            else -> null
        }
    }
}
